packages/store/tests/selectors/create-model-selector.spec.ts

```diff
@@ -0,0 +1,262 @@
+  });
+
+  it('should create a model from the selectors in the selector map', () => {
+    // Arrange
+    const { store, stateSelector, setState } = setupFixture();
+    setState({ property1: 'Tada', property2: [1, 3, 5], property3: { hello: 'there' } });
+    const props = createPropertySelectors(stateSelector);
+    const prop3 = createPropertySelectors(props.property3);
+    // Act
+    const modelSelector = createModelSelector({
+      foo: props.property2,
+      bar: props.property1,
+      hello: prop3.hello,
+    });
+    // Assert
+    expect(modelSelector).toBeDefined();
+    expect(store.selectSnapshot(modelSelector)).toStrictEqual({
+      foo: [1, 3, 5],
+      bar: 'Tada',
+      hello: 'there',
+    });
+  });
+
+  describe('[memoization]', () => {
+    it('should change if a specified property changes', () => {
+      // Arrange
+      const { store, stateSelector, setState, patchState } = setupFixture();
+      setState({ property1: 'Tada', property2: [1, 3, 5], property3: { hello: 'there' } });
+      const props = createPropertySelectors(stateSelector);
+      const prop3 = createPropertySelectors(props.property3);
+      // Act
+      const modelSelector = createModelSelector({
+        foo: props.property2,
+        bar: props.property1,
+        hello: prop3.hello,
+      });
+      // Assert
+      expect(modelSelector).toBeDefined();
+      const snapshot1 = store.selectSnapshot(modelSelector);
+      expect(snapshot1).toStrictEqual({ foo: [1, 3, 5], bar: 'Tada', hello: 'there' });
+      patchState({ property1: 'Hi' });
+      const snapshot2 = store.selectSnapshot(modelSelector);
+      expect(snapshot2).not.toBe(snapshot1);
+      expect(snapshot2).toStrictEqual({ foo: [1, 3, 5], bar: 'Hi', hello: 'there' });
+      patchState({ property2: [2, 4] });
+      const snapshot3 = store.selectSnapshot(modelSelector);
+      expect(snapshot3).not.toBe(snapshot1);
+      expect(snapshot3).not.toBe(snapshot2);
+      expect(snapshot3).toStrictEqual({ foo: [2, 4], bar: 'Hi', hello: 'there' });
+    });
+
+    it('should not change if an unspecified property changes', () => {
+      // Arrange
+      const { store, stateSelector, setState, patchState } = setupFixture();
+      setState({ property1: 'Tada', property2: [1, 3, 5], property3: { hello: 'there' } });
+      const props = createPropertySelectors(stateSelector);
+      const prop3 = createPropertySelectors(props.property3);
+      // Act
+      const modelSelector = createModelSelector({
+        bar: props.property1,
+        hello: prop3.hello,
+      });
+      // Assert
+      expect(modelSelector).toBeDefined();
+      const snapshot1 = store.selectSnapshot(modelSelector);
+      expect(snapshot1).toStrictEqual({ bar: 'Tada', hello: 'there' });
+      patchState({ property2: [2, 4] });
+      const snapshot2 = store.selectSnapshot(modelSelector);
+      expect(snapshot2).toBe(snapshot1);
+      expect(snapshot1).toStrictEqual({ bar: 'Tada', hello: 'there' });
```

Comment:
For memoisation should we test that `this.store.select` observable does not emit?

Reply:
This is implied because the snapshot returns the exact same object (same reference).
`select` has a `distinctUntilChanged` so it would not emit.
This is just the simpler way to test this.